data/hydraulic-data.py

```diff
@@ -0,0 +1,147 @@
+                "stable_y": pd.DataFrame(pd_frame.iloc[:,4])
+            }
+            pd_target_label_y = pd_target_labels_y.get(target_label, None)
+            pd_target_label_y.columns = [target_label]
+            return pd_target_label_y
+
+        def extract_hydraulic_zip(file: str, output_file: str, target_label: str) -> None:
+            """Read zip file, extract hydraulic sensor data fields and labels, store into csv
+
+            Keyword arguments:
+            file -- filepath to zip file represented as a string
+            output_file -- destination file where hydraulic processed data is saved
+            target_label -- chosen hydraulic system condition monitoring label for supervised learning
+            """
+            with zipfile.ZipFile(file,"r") as zip:
+                # import all data of pressure sensors
+                df_ps1 = import_data_field(zip, "PS1.txt", "psa_bar")
+                df_ps2 = import_data_field(zip, "PS2.txt", "psb_bar")
+                df_ps3 = import_data_field(zip, "PS3.txt", "psc_bar")        
+                df_ps4 = import_data_field(zip, "PS4.txt", "psd_bar")
+                df_ps5 = import_data_field(zip, "PS5.txt", "pse_bar")
+                df_ps6 = import_data_field(zip, "PS6.txt", "psf_bar")
+
+                # import all data of volume flow sensors
+                df_vf1 = import_data_field(zip, "FS1.txt", "fsa_vol_flow")
+                df_vf2 = import_data_field(zip, "FS2.txt", "fsb_vol_flow")
+
+                # import all data of temperature sensors
+                df_ts1 = import_data_field(zip, "TS1.txt", "tsa_temp")
+                df_ts2 = import_data_field(zip, "TS2.txt", "tsb_temp")
+                df_ts3 = import_data_field(zip, "TS3.txt", "tsc_temp")
+                df_ts4 = import_data_field(zip, "TS4.txt", "tsd_temp")
+
+                # import data of pump efficiency sensor
+                df_pe1 = import_data_field(zip, "EPS1.txt", "pump_eff")
+
+                # import data of vibrations sensor
+                df_vs1 = import_data_field(zip, "VS1.txt", "vs_vib")
+
+                # import data of cooling efficiency sensor
+                df_ce1 = import_data_field(zip, "CE.txt", "cool_eff_pct")
+
+                # import data of cooling power sensor
+                df_cp1 = import_data_field(zip, "CP.txt", "cool_pwr_pct")
+
+                # import data of efficiency factor sensor
+                df_ef1 = import_data_field(zip, "SE.txt", "eff_fact_pct")
+
+                # import one of the hydraulic condition monitoring labels from profile based on target label
+                df_hyd_cond_monitor_label_y = import_target_label(zip, "profile.txt", target_label)        
+                
+                # Combine all sensor Dataframes and then add hydraulic condition monitoring label as last column
+                df_hyd_cond_monitor_data_x = pd.concat([df_ps1, df_ps2, df_ps3, df_ps4, df_ps5, df_ps6, df_vf1,
```

Comment:
These comments are great 👍 - thanks!   Made the code easy to glance through for me while not having to be familiar with the name of the files in the zip. 🤩